### ISSUE TITLE
Pycharm freezing while debugging(both fast and normal)

### ISSUE THREAD

**Issue by ArthurNRL** [reloadware/reloadium#143]
## Describe the bug*
Pucharm freezes while using Reloadium debug mode, both fast and normal

## To Reproduce
Not really sure if its my project but the steps are simple
1. Start debugging
2. Do some changes
3. See changes be reloaded
4. After some time pycharm just freezes on step line or continue

## Expected behavior
Pycharm doesn't freeze

## Desktop or remote (please complete the following information):**
 - OS: Windows
 - OS version: 11
 - PyCharm plugin version: 1.1.0
 - Editor: PyCharm
 - Python Version: 3.11.0
 - Python Architecture: 64bit
 - Run mode: Debug/Fast Debug

## Additional context
Got this log from idea.log
```
2023-04-26 15:44:58,739 [ 501376] SEVERE - #c.i.c.c.CompletionProgressIndicator - class rw.session.cmds.Cmd$Return cannot be cast to class rw.session.cmds.completion.GetCtxCompletion$Return (rw.session.cmds.Cmd$Return and rw.session.cmds.completion.GetCtxCompletion$Return are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @7ad9ec8b)
java.lang.ClassCastException: class rw.session.cmds.Cmd$Return cannot be cast to class rw.session.cmds.completion.GetCtxCompletion$Return (rw.session.cmds.Cmd$Return and rw.session.cmds.completion.GetCtxCompletion$Return are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @7ad9ec8b)
	at rw.completion.CtxCompletionContributor$Provider.addCompletions(CtxCompletionContributor.java:86)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:23)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at rw.completion.BaseCompletionContributor.fillCompletionVariants(BaseCompletionContributor.java:35)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributor(CompletionService.java:81)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.lambda$getVariantsFromContributor$0(CompletionServiceImpl.java:333)
	at com.intellij.diagnostic.telemetry.TraceKt.runWithSpan(trace.kt:74)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.getVariantsFromContributor(CompletionServiceImpl.java:332)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:73)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:152)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:144)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:140)
	at com.jetbrains.python.codeInsight.completion.PyModuleNameCompletionContributor.fillCompletionVariants(PyModuleNameCompletionContributor.kt:48)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributor(CompletionService.java:81)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.lambda$getVariantsFromContributor$0(CompletionServiceImpl.java:333)
	at com.intellij.diagnostic.telemetry.TraceKt.runWithSpan(trace.kt:74)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.getVariantsFromContributor(CompletionServiceImpl.java:332)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:73)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:152)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:144)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:140)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:87)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:23)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributor(CompletionService.java:81)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.lambda$getVariantsFromContributor$0(CompletionServiceImpl.java:333)
	at com.intellij.diagnostic.telemetry.TraceKt.runWithSpan(trace.kt:74)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.getVariantsFromContributor(CompletionServiceImpl.java:332)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:73)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:56)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:132)
	at com.intellij.codeInsight.completion.BaseCompletionService.performCompletion(BaseCompletionService.java:48)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.lambda$performCompletion$1(CompletionServiceImpl.java:351)
	at com.intellij.diagnostic.telemetry.TraceKt.runWithSpan(trace.kt:74)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.performCompletion(CompletionServiceImpl.java:340)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.lambda$calculateItems$11(CompletionProgressIndicator.java:870)
	at com.intellij.util.indexing.FileBasedIndex.lambda$ignoreDumbMode$0(FileBasedIndex.java:199)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:112)
	at com.intellij.util.indexing.FileBasedIndexEx.ignoreDumbMode(FileBasedIndexEx.java:719)
	at com.intellij.util.indexing.FileBasedIndex.ignoreDumbMode(FileBasedIndex.java:198)
	at com.intellij.util.indexing.DumbModeAccessType.ignoreDumbMode(DumbModeAccessType.java:43)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:866)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:854)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$7(CodeCompletionHandlerBase.java:374)
	at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:212)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:169)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1102)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:167)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$8(CodeCompletionHandlerBase.java:365)
	at io.opentelemetry.context.Context.lambda$wrap$1(Context.java:212)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$0(CompletionThreading.java:92)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:88)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:252)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
2023-04-26 15:44:58,741 [ 501378] SEVERE - #c.i.c.c.CompletionProgressIndicator - PyCharm 2023.1  Build #PY-231.8109.197
2023-04-26 15:44:58,742 [ 501379] SEVERE - #c.i.c.c.CompletionProgressIndicator - JDK: 17.0.6; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.
2023-04-26 15:44:58,742 [ 501379] SEVERE - #c.i.c.c.CompletionProgressIndicator - OS: Windows 11
2023-04-26 15:44:58,742 [ 501379] SEVERE - #c.i.c.c.CompletionProgressIndicator - Plugin to blame: Reloadium - ChatGPT, Hot Reloading and Profiling for Python version: 1.1.0
```
I also have some thread freeze dumps if you need it

**Comment by dkrystki** [comment]
@ArthurNRL Could you provide your PyCharm version?

**Comment by ArthurNRL** [comment]
2023.1, using new ui if it helps

**Comment by dkrystki** [comment]
While I'm investigating this could you disable Runtime completion and try again?
![2023-04-29_13:32:05](https://user-images.githubusercontent.com/45499236/235281537-81918e50-01b0-4b33-b905-9f7305a64717.png)

**Comment by ArthurNRL** [comment]
I'll be testing this throughout the week

**Comment by ArthurNRL** [comment]
@dkrystki it hasn't crashed on me yet

**Comment by dkrystki** [comment]
Fixed in Reloadium 1.1.1 and PyCharm plugin 1.1.1

**Comment by gleb-sevruk** [comment]
It looks like still freezing
2023-07-21 01:36:55,000 [10240515]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1984 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:36:56,552 [10242067]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:36:56,552 [10242067]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:36:57,006 [10242521]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 2005 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.getAccessibleRole(CAccessibility.java:286)
2023-07-21 01:36:59,007 [10244522]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1639 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:37:00,768 [10246283]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:37:00,769 [10246284]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 12736ms, details saved to /Users/gleb/Library/Logs/JetBrains/PyCharm2023.1/threadDumps-freeze-20230721-013653-PY-231.9225.15-12sec
2023-07-21 01:37:00,772 [10246287]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-07-21 01:37:06,055 [10251570]   INFO - #c.i.o.d.Logger - Plots on port: 63342
2023-07-21 01:37:06,077 [10251592]   INFO - #rw.handler.PythonRunConfHandler - Deactivating dbnd_azure_datafactory_monitor
2023-07-21 01:37:26,950 [10272465]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-07-21 01:38:09,652 [10315167]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:38:09,756 [10315271]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:38:12,291 [10317806]   INFO - #rw.session.Client - Client connected to the Session
2023-07-21 01:38:12,293 [10317808]   INFO - #rw.session.events.Handshake - Handshake from the client, version = 1.2.0
2023-07-21 01:38:13,002 [10318517]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1184 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:38:13,971 [10319486]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:38:13,971 [10319486]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:38:15,003 [10320518]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 2001 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.getAccessibleRole(CAccessibility.java:286)
2023-07-21 01:38:17,004 [10322519]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1143 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:38:18,173 [10323688]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:38:18,174 [10323689]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:38:19,007 [10324522]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 2003 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.getAccessibleRole(CAccessibility.java:286)
2023-07-21 01:38:21,000 [10326515]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1976 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:38:22,390 [10327905]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:38:22,390 [10327905]   INFO - #rw.session.Client - Sending cmd ReloadFile
2023-07-21 01:38:23,005 [10328520]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 2005 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.getAccessibleRole(CAccessibility.java:286)
2023-07-21 01:38:25,001 [10330516]   INFO - sun.awt.AWTThreading - InvocationEvent has timed out (awaiting 1989 ms). Originated at java.desktop/sun.lwawt.macosx.CAccessibility.accessibilityHitTest(CAccessibility.java:481)
2023-07-21 01:38:26,596 [10332111]   INFO - #rw.session.Client - Timeout waiting for "ReloadFile" return
2023-07-21 01:38:26,597 [10332112]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 16946ms, details saved to /Users/gleb/Library/Logs/JetBrains/PyCharm2023.1/threadDumps-freeze-20230721-013814-PY-231.9225.15-16sec
